src/WholeBodyStateDisplay.cpp

```diff
@@ -636,9 +733,15 @@ void WholeBodyStateDisplay::processWholeBodyState() {
       if (grf_enable_ && active_contact_in_grf) {
         boost::shared_ptr<ArrowVisual> arrow;
         arrow.reset(new ArrowVisual(context_->getSceneManager(), scene_node_));
-        arrow->setArrow(contact_pos, contact_for_orientation);
-        arrow->setFramePosition(position);
-        arrow->setFrameOrientation(orientation);
+        if (grf_locate_at_foot_cop_ && foot_cop_enable_ && active_contact_in_foot_cop && is_contact_6d) {
+          arrow->setArrow(foot_cop_pos_[i], contact_for_orientation);
+          arrow->setFramePosition(contact_pos);
+          arrow->setFrameOrientation(orientation);
+        } else {
+          arrow->setArrow(contact_pos, contact_for_orientation);
+          arrow->setFramePosition(position);
+          arrow->setFrameOrientation(orientation);
+        }
 
```

Comment:
This if-else decides whether to plot the GRFs at the per foot CoP, or at the contact origin. The final three booleans in the if are pulled straight from the per foot CoP logic, so this should only evaluate to true if the per foot CoP has also been calculated. Otherwise, plot at contact origin.